src/Uno.UI/UI/Xaml/DependencyPropertyDetailsCollection.Bindings.cs

```diff
@@ -14,8 +14,8 @@ namespace Windows.UI.Xaml
 	/// </summary>
 	partial class DependencyPropertyDetailsCollection
 	{
-		private ImmutableList<BindingExpression> _bindings = ImmutableList<BindingExpression>.Empty;
-		private ImmutableList<BindingExpression> _templateBindings = ImmutableList<BindingExpression>.Empty;
+		private List<BindingExpression> _bindings;
```

Comment:
Let's make sure to add a test for validating that the adding/removing bindings during binding enumeration does not break, as it was the original intent behind the immutable lists. E.g. in this new implementation, a newly added binding may be processed in the same loop, where it previously wasn't.